apps/studio/components/interfaces/Settings/Addons/ComputeInstanceSidePanel.tsx

```diff
@@ -123,32 +101,79 @@ const ComputeInstanceSidePanel = () => {
       })
     },
   })
+
+  const [selectedOption, setSelectedOption] = useState<string>('')
+
   const isSubmitting = isUpdating || isRemoving
 
   const projectId = selectedProject?.id
   const cpuArchitecture = getCloudProviderArchitecture(selectedProject?.cloud_provider)
   const selectedAddons = addons?.selected_addons ?? []
-  const availableAddons = addons?.available_addons ?? []
+  const availableAddons = useMemo(() => {
+    return addons?.available_addons ?? []
+  }, [addons])
 
   const isFreePlan = subscription?.plan?.id === 'free'
   const subscriptionCompute = selectedAddons.find((addon) => addon.type === 'compute_instance')
   const pitrAddon = selectedAddons.find((addon) => addon.type === 'pitr')
-  const availableOptions =
-    availableAddons.find((addon) => addon.type === 'compute_instance')?.variants ?? []
+
+  const hasMicroOptionFromApi = useMemo(() => {
+    return (
+      availableAddons.find((addon) => addon.type === 'compute_instance')?.variants ?? []
+    ).some((variant) => variant.identifier === 'ci_micro')
+  }, [availableAddons])
+
+  const availableOptions = useMemo(() => {
+    const computeOptions =
+      availableAddons.find((addon) => addon.type === 'compute_instance')?.variants ?? []
+
+    // Backwards comp until API is deployed
+    if (!hasMicroOptionFromApi) {
+      // Unshift to push to start of array
+      computeOptions.unshift({
+        identifier: 'ci_micro',
+        name: 'Micro',
+        price_description: '$0.01344/hour (~$10/month)',
+        price: 0.01344,
+        price_interval: 'hourly',
+        price_type: 'usage',
+        // @ts-ignore API types it as Record<string, never>
+        meta: {
+          cpu_cores: 2,
+          cpu_dedicated: false,
+          memory_gb: 1,
+          baseline_disk_io_mbs: 87,
+          max_disk_io_mbs: 2085,
+          connections_direct: 60,
+          connections_pooler: 200,
+        } as ProjectAddonVariantMeta,
```

Comment:
Looks like there are 3 instances of these, or a subset thereof, should there be a constant?

Reply:
I may be missing something, why not `meta: INSTANCE_MICRO_SPECS`, since in the end it's still typed as `Record<string, never>`? Relatedly, it may make sense for `ProjectAddonVariantMeta` to extend `Partial<InstanceSpecs>`, since any instance specs could be considered to be metadata for compute addons, but I understand why we may want to keep them decoupled too.